From 71b32fb6b36188c699b169af49a3ddb2f71bddf3 Mon Sep 17 00:00:00 2001
From: bashonly <88596187+bashonly@users.noreply.github.com>
Date: Sun, 22 Oct 2023 12:58:53 +0300
Subject: [PATCH] Revert "fftools/ffmpeg_mux_init: drop a duplicated block in
 copy_metadata()"

This reverts commit 3c7dd5ed37da6d2de06c4850de5a319ca9cdd47f.
---
 fftools/ffmpeg_mux_init.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/fftools/ffmpeg_mux_init.c b/fftools/ffmpeg_mux_init.c
index e42d5ff26a..1a9611d19d 100644
--- a/fftools/ffmpeg_mux_init.c
+++ b/fftools/ffmpeg_mux_init.c
@@ -1822,6 +1822,16 @@ static int copy_metadata(Muxer *mux, AVFormatContext *ic,
     parse_meta_type(mux, inspec,  &type_in,  &idx_in,  &istream_spec);
     parse_meta_type(mux, outspec, &type_out, &idx_out, &ostream_spec);
 
+    if (!ic) {
+        if (type_out == 'g' || !*outspec)
+            *metadata_global_manual = 1;
+        if (type_out == 's' || !*outspec)
+            *metadata_streams_manual = 1;
+        if (type_out == 'c' || !*outspec)
+            *metadata_chapters_manual = 1;
+        return 0;
+    }
+
     if (type_in == 'g' || type_out == 'g')
         *metadata_global_manual = 1;
     if (type_in == 's' || type_out == 's')
-- 
2.39.2

